server/src/main/java/org/opensearch/index/codec/composite/Composite99DocValuesWriter.java

```diff
@@ -75,6 +76,10 @@ public Composite99DocValuesWriter(DocValuesConsumer delegate, SegmentWriteState
     @Override
     public void addNumericField(FieldInfo field, DocValuesProducer valuesProducer) throws IOException {
         delegate.addNumericField(field, valuesProducer);
+        // Perform this only during flush flow
+        if (mergeState.get() == null && segmentHasCompositeFields) {
```

Comment:
Do we need any additional handling in ctor to create segmentHasCompositeFields since check today is based on SORTED_NUMERIC?

Reply:
Yeah really good point. Thanks will add.